woocommerce/woocommerce-gutenberg-products-block
# Fix button style of Mini Cart block

Fixes #4679

This PR style the mini cart button to match the design. A new component named `QuantityBadge` was added to insolate the style of the badge. On screens smaller than 768px, only the badge with the count is displayed. On bigger screens, cart total, count, and the cart icon (without the counter) are displayed.

Update [October 28th]: Working on this I found two bugs of `formatPrice` function, please see the screenshots below or check two issues: #5018 and #5019.




#### Accessibility




### Screenshots
**Editor:**

![image](https://user-images.githubusercontent.com/5423135/139178205-8743bd5e-2016-40f0-be22-3df09ca26511.png)

**Front-end - full:**

![image](https://user-images.githubusercontent.com/5423135/139177945-d58cf908-0635-4bd3-ab57-554242f18ead.png)

**Front-end - full - after lazy loading:**

![image](https://user-images.githubusercontent.com/5423135/139178702-bce6fe0f-57a0-443b-b0b3-55ce657006a2.png)

**Front-end - empty:**

![image](https://user-images.githubusercontent.com/5423135/139178031-93fed97e-8bb2-448c-98c2-2350d9e11bd9.png)

**Front-end - empty - after lazy loading:**

![image](https://user-images.githubusercontent.com/5423135/139178091-10029de6-16fd-403d-8fa2-be64d54212c4.png)

**Smaller screens (<768px):**

![image](https://user-images.githubusercontent.com/5423135/139178258-6de651bb-2819-41b0-a40c-682a67cdeb40.png)

### Testing
1. Add a columns block.
2. Add Mini Cart Block into a column.
3. See the design changes: the total, number of items, and the cart icon as the screenshots above.
4. Change the browser width to less than `768px`, see the layout change to only cart icon with a counter badge.

### Changelog

> Fix button style of Mini Cart block for small screens.
